cartridge-cli.lua

```diff
@@ -2385,6 +2378,52 @@ local function check_if_deprecated_build_flow_is_ised()
     return deprecated_build_flow_is_ised
 end
 
+-- * ------------------- Build dir --------------------
+
+local BUILD_SUBDIRECTORY_NAME = 'cartridge.build'
+
+local function detect_and_check_build_dir()
+    -- By default, application is built in the tmpdir.
+    -- User can specify build directory in TARANTOOL_BUILDDIR env variable.
+    -- There are two cases:
+    -- - specified directory doesn't exists: we just create it and remove after the build
+    -- - directory already exists:
+    --   - ${TARANTOOL_BUILDDIR}/cartridge.build will be the build directory
+    --   - sub-directory cartridge.build is (re)created and used for application build
+    --   - after the build, ${TARANTOOL_BUILDDIR}/cartridge.build  is removed
+
+    local specified_build_dir = os.getenv('TARANTOOL_BUILDDIR')
+
+    if specified_build_dir == nil then
+        -- just use tmpdir
+        return fio.tempdir()
+    end
+
+    local build_dir = fio.abspath(specified_build_dir)
+
+    if not fio.path.exists(build_dir) then
+        -- create this diectory and use it
+        fio.mktree(build_dir)
+        return build_dir
+    end
+
+    if not fio.path.is_dir(build_dir) then
+        die("Specified build directory is not a directory: %s", build_dir)
+    end
+
+    -- This little hack is used to prevent deletion user files
+    -- from specified build directory on cleanup.
+    -- Moreover, this subdirectory is defenitely clean,
+    -- so we wouldn't have any problems
+    build_dir = fio.pathjoin(build_dir, BUILD_SUBDIRECTORY_NAME)
+    fio.rmtree(build_dir)
```

Comment:
This may be extremely unexpected for the user - let's log this action. `Directory is already exists - clean it`